collect/src/main/kotlin/com/certora/collect/EmptyTreapList.kt

```diff
@@ -0,0 +1,66 @@
+package com.certora.collect
+
+internal class EmptyTreapList<E> private constructor() : TreapList<E>, java.io.Serializable, AbstractList<E>() {
```

Comment:
that's an interesting way of doing this...
I know the standard approach - that there is the recursive data structure, which if empty is null, but it's not really exposed, and always wrapped in an outer box. 

This approach saves the memory needed for the outer object?

Reply:
It saves the additional GC heap churn from allocating the outer object every time we generate a new list.  It also lets us optimize specifically for operations on empty lists (though who knows how significant that is?).